Move content from the GitHub Wiki to the Hub Vault

claremacrae opened the pull request:
<!-- Add a small description here of the changes you added -->

- Migrate almost all content from the [Hub wiki](https://github.com/obsidian-community/obsidian-hub/wiki) to the Hub Vault
- The changes marked as co-authored by @argenos were done via pairing, so reviewed at the time. (Further review welcomed!)
- Changes after that were done by me alone

Note that we have not added descriptive text about the structure of the new folders and files. This can be done as a later change.
- Added/Edited note ...

## Checklist

- [x] I have renamed all attached images with descriptive file names (or I did not include any images)
- [x] Before creating a new note, I searched the vault (or I only edited existing notes)
- [ ] **Not done, due to time constraints** (Optional) In case I created a new note in the folder `04 - Guides, Workflows, & Courses`, I added a link to the new note(s) in one of the `for {group X}` overviews ([listed here](https://publish.obsidian.md/hub/04+-+Guides%2C+Workflows%2C+%26+Courses/%F0%9F%97%82%EF%B8%8F+04+-+Guides%2C+Workflows%2C+%26+Courses)).

claremacrae commented:
I've just noticed that a .DS_Store file has been created.

I really don't understand this - as it's supposed to be being ignored...

https://github.com/obsidian-community/obsidian-hub/blob/main/.gitignore#L8

claremacrae commented:
> I've just noticed that a `.DS_Store` file has been created.
> 
> I really don't understand this - as it's supposed to be being ignored...
> 
> https://github.com/obsidian-community/obsidian-hub/blob/main/.gitignore#L8

It turns out that the file was already present in git, which is why the .gitignore setting didn't matter.

I have used  `git rm` to remove it.

kometenstaub commented on 00 - Contribute to the Obsidian Hub/03 Contributor Notes/03.02 Design Decisions/Content Comments.md:
```diff
@@ -0,0 +1,54 @@
+
+## Agree a standard begin/end prefix for marking 'do not edit' sections
+
+We use these markers in Author pages, and they do not seem to be well-understand:
```
```suggestion
We use these markers in Author pages, and they do not seem to be well-understood:
```

kometenstaub commented on 00 - Contribute to the Obsidian Hub/03 Contributor Notes/03.02 Design Decisions/Content Lifecycle of Extensions.md:
```diff
@@ -0,0 +1,87 @@
+- **Published**
+    - Is currently in community plugins file
+    - This means that `update-releases.py` will create/update a hub page for the plugin
+    - Repo must not be archived not archived (but we don't check that currently)
```
```suggestion
    - Repo must not be archived (but we don't check that currently)
```

kometenstaub commented on 00 - Contribute to the Obsidian Hub/03 Contributor Notes/03.02 Design Decisions/Content Lifecycle of Extensions.md:
```diff
@@ -0,0 +1,87 @@
+
+- **The plugin's GitHub**
+    - Currently used:
+        - Various details like author name, description
```
```suggestion
        - Various details like author, name, description
```

claremacrae replied:
This really does mean "name of author"

claremacrae replied:
Have changed it to `author's name` to remove the ambiguity

kometenstaub commented on 00 - Contribute to the Obsidian Hub/03 Contributor Notes/03.02 Design Decisions/Content Lists.md:
```diff
@@ -0,0 +1,40 @@
+
+### Decisions and conclusions
+
+- Sort on Alias, not Plugin ID
```
```suggestion
- Sort by Alias, not Plugin ID
```
I think it should be "sort *by*"

kometenstaub commented on 00 - Contribute to the Obsidian Hub/03 Contributor Notes/03.02 Design Decisions/Content Lists.md:
```diff
@@ -0,0 +1,40 @@
+### Decisions and conclusions
+
+- Sort on Alias, not Plugin ID
+- Avoid comment markers
```
What do you mean by comment markers?

claremacrae replied:
Wording improved in 2ff0699c58904a2c298e6bfce9192916d3b9b43c

kometenstaub commented on 00 - Contribute to the Obsidian Hub/03 Contributor Notes/03.02 Design Decisions/Content Lists.md:
```diff
@@ -0,0 +1,40 @@
+
+For human-edited lists, e.g. categories:
+
+Introduce a standard delimiter to mark lists that should be sorted by the text the link, case-insensitively, e.g. 
```
> The text the link

claremacrae replied:
Fixed in 36d9bf90002f073c87f65accb9f36edba5d65be9

kometenstaub commented on 00 - Contribute to the Obsidian Hub/03 Contributor Notes/03.03 Scripts and Automation/Testing Python Code with Approval Tests.md:
```diff
@@ -0,0 +1,88 @@
+1. Fix the code, if the differences show an unwanted change
+2. Use the diff tool to copy the changes over from received to approved, if the changes are intentional.
+
+### Scenario: First run of a new test
```
This is a level 3 heading without any text below it. Should the following headings be level 4 instead of level 3?

claremacrae replied:
Missing text added in f58865e70e692c1cacba64164631133341e593c2

kometenstaub commented on 00 - Contribute to the Obsidian Hub/03 Contributor Notes/03.03 Scripts and Automation/Testing Python Code with Approval Tests.md:
```diff
@@ -0,0 +1,88 @@
+
+### Diff Tools (a.k.a. "Reporters")
+
+**Note**: to be used effectively, Approval Tests require you to have a diff tool installed.
```
For Vim users, they could use vimdiff https://vi.stackexchange.com/a/626

claremacrae replied:
It's not in the list of diff tools currently supported, and right now I don't think it's trivial to use custom reporters (or rather, I haven't been able to figure out how to do so) so I think adding this might cause confusion.

kometenstaub replied:
I see, I should have followed the link below. :sweat_smile:

kometenstaub commented on 00 - Contribute to the Obsidian Hub/03 Contributor Notes/03.03 Scripts and Automation/Updating MOC files.md:
````diff
@@ -0,0 +1,12 @@
+
+```bash
+cd obsidian-hub/.github/scripts
+./update-mocs.py
````
```suggestion
python3 ./update-mocs.py
```
Like above it should include python3 so that Windows users can run it without issues.

kometenstaub commented on 00 - Contribute to the Obsidian Hub/Obsidian Settings for Contributors.md:
```diff
@@ -0,0 +1,17 @@
+
+## Argentum's Settings
+
+Argentum has kindly provided a Zipped copy of the `.obsidian` directory from a properly-configured hub vault.
```
I think this can be removed because of line 15.

claremacrae replied:
Fixed in a1be56190c8caa6635bb70836d157db79f57a106

kometenstaub reviewed (changes requested):
I read through it on github and mostly looked at duplicated words/errors. I didn't check whether all the links are working.

claremacrae commented:
I just discovered that I didn't bring in screenshots and images from the vault. Fixing it now...

claremacrae commented:
> I read through it on github and mostly looked at duplicated words/errors. I didn't check whether all the links are working.
> 
> Where I suggested changes, you can apply them directly by clicking on "commit suggestion".

Hi @kometenstaub - these were really helpful comments, thank you very much!

I believe I have addressed all of your comments. Where they were trivial edits, I've just marked the conversation as resolved.

For others, I've put in a link to the commit where I addressed the comment, in case you were inclined to review the fixes.

Thanks again!

claremacrae commented:
I've gone ahead and merged this, as much of it was done via pairing with @argenos - and @kometenstaub has reviewed it too. Thanks both! :-)
